decorator factory.py

```diff
@@ -0,0 +1,60 @@
+from functools import wraps
+from typing import Callable
+"""
+The code is a decorator factory.
+The in the factory decorator receives an argument type, and validates in the func call, that the received
+argument is from the defined type.
+If it is, the factory returns the decorated function.
+Else, it raise a TypeError exception.
+"""
+
+
+def decorator_factory(correct_type: type) -> Callable:
```

Comment:
Dont name it factory, the fact that it is a factory is not the most important thing about it. Name it according to what it does